python/cudf/cudf/io/parquet.py

```diff
@@ -175,29 +175,47 @@ def read_parquet(
 ):
     """{docstring}"""
 
-    filepath_or_buffer, compression = ioutils.get_filepath_or_buffer(
-        filepath_or_buffer, None, **kwargs
-    )
-    if compression is not None:
-        raise ValueError("URL content-encoding decompression is not supported")
+    # Multiple sources are passed as a list. If a single source is passed,
+    # wrap it in a list for unified processing downstream.
+    if not is_list_like(filepath_or_buffer):
+        filepath_or_buffer = [filepath_or_buffer]
+
+    # a list of row groups per source should be passed. make the list of
+    # lists that is expected for multiple sources
+    if row_groups is not None:
+        if not is_list_like(row_groups):
+            row_groups = [[row_groups]]
+        elif not is_list_like(row_groups[0]):
+            row_groups = [row_groups]
+
+    filepaths_or_buffers = []
+    for source in filepath_or_buffer:
+        tmp_source, compression = ioutils.get_filepath_or_buffer(
+            source, None, **kwargs
+        )
```

Comment:
This can be handled in a followup pr: The current logic doesn't work when the user passes a filepath with a wildcard like `path/*.parquet` since we call the `ioutils.get_filepath_or_buffer` method on each string and the logic implemented [here](https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/utils/ioutils.py#L934-L940) guards against passing multiple files (since it was unsupported at the time). This utility can potentially be updated to return a list of files and readers which support this use it while others warn/raise errors.